packages/core/src/_common/base/api_middleware.js

```diff
@@ -20,19 +105,37 @@ class APIMiddleware {
         const promise = promiseRejectToResolve(response_promise);
 
         const key = requestToKey(request);
+        const request_type = this.getRequestType(request);
+
+        if (request_type) performance.mark(`${request_type}_start`);
 
         if (options.callback) {
             promise.then(options.callback);
         }
 
         this.debounced_calls[key] = promise;
 
-        promise.then(() => {
+        promise.then(response => {
+            const response_type = this.getRequestType(response);
+            if (response_type) this.defineMeasure(response_type);
+
```

Comment:
if we can't have a proxy then can we add the p2p performance measures related logic on the condition of pathname if it is p2p, but here too the condition will be checked for every request in app but at least the logic won't be executed , or any other way we can achieve it ?

Reply:
there is no need for this as i have restricted the logs to be sent only when the user is in p2p